build.rs

```diff
@@ -46,77 +46,27 @@ fn bindgen_headers() -> Result<()> {
 fn main() -> Result<()> {
     bindgen_headers()?;
 
-    // Check whether we can use 64-bit compilation
-    let use_64bit_compilation = if env::var("CARGO_CFG_TARGET_POINTER_WIDTH").unwrap() == "64" {
-        let check = cc::Build::new()
-            .file("depend/check_uint128_t.c")
-            .cargo_metadata(false)
-            .try_compile("check_uint128_t")
-            .is_ok();
-        if !check {
-            println!("cargo:warning=Compiling in 32-bit mode on a 64-bit architecture due to lack of uint128_t support.");
-        }
-        check
-    } else {
-        false
-    };
     let target = env::var("TARGET").expect("TARGET was not set");
-    let is_big_endian = env::var("CARGO_CFG_TARGET_ENDIAN").expect("No endian is set") == "big";
     let mut base_config = cc::Build::new();
 
+    language_std(&mut base_config, "c++17");
+
     base_config
-        .cpp(true)
         .include("depend/zcash/src")
         .include("depend/zcash/src/rust/include/")
         .include("depend/zcash/src/secp256k1/include")
         .flag_if_supported("-Wno-implicit-fallthrough")
         .flag_if_supported("-Wno-catch-value")
         .flag_if_supported("-Wno-reorder")
         .flag_if_supported("-Wno-deprecated-copy")
+        .flag_if_supported("-Wno-unused-parameter")
+        .flag_if_supported("/wd4100")
```

Comment:
What does this warning do?
Can we add a comment here?

```suggestion
        // when compiling using Microsoft Visual C++, ignore warnings about unused arguments
        .flag_if_supported("/wd4100")
```